hack/devtools/msi.sh

```diff
@@ -3,20 +3,17 @@
 # This script creates a mock MSI object to use for local development
 # We use a service principal and certificate as the mock object
 
-appName="mock-msi-$(openssl rand -base64 9 | tr -dc 'a-zA-Z0-9' | head -c 6)"
+scriptPath=$(realpath "$0")
+scriptDir=$(dirname "$scriptPath")
 
-result=$(az ad sp create-for-rbac --name $appName --create-cert)
+source "$scriptDir/local_dev_env.sh"
 
-# Extract the client ID and tenant ID from the result
-clientID=$(echo $result | jq -r .appId)
-tenantID=$(echo $result | jq -r .tenant)
-
-# Extract the certificate information from the result
-certFilePath=$(echo $result | jq -r '.fileWithCertAndPrivateKey')
-base64EncodedCert=$(base64 -w 0 $certFilePath)
-rm $certFilePath
+sp=$(create_mock_msi)
```

Comment:
Doesn't want to hold you from merging this since it's for Dev. But you can add err check or sp is not nil condition.

Reply:
I'll add this when we merge to master.